code/API_definitions/Application_Endpoint_Discovery_API.yaml

```diff
@@ -0,0 +1,596 @@
+    ```
+    Example 2: Request with IP-Address
+    ```
+      GET /apps-endpoints?filter=closest HTTP/1.1
+      Host: example.com
+      Authorization: Bearer {your_access_token}
+      Accept: application/json
+      IP-Address: 84.125.93.10
+    ```
+    Example 3: Request with Phone-Number
+    ```
+      GET /apps-endpoints?filter=closest HTTP/1.1
+      Host: example.com
+      Authorization: Bearer {your_access_token}
+      Accept: application/json
+      Phone-Number: +441234567890
+    ```
+    # Authorization and authentication
+    In order to maintain the integrity and security of the Application Discovery
+    API, access is governed by robust authentication and authorization
+    processes.
+    This ensures that only valid, authorized requests are able to interact with
+    the API and retrieve application endpoint information.
+    ## OpenID Connect
+    The API utilizes OpenID Connect on top of OAuth 2.0 for user authentication.
+    Users must register to obtain client credentials, which are necessary to
+    acquire access tokens from the OpenID provider.
+    ## Obtaining Access Tokens
+    Access tokens are obtained by submitting a request to the designated OpenID
+    Connect URL, utilizing the client credentials. It is imperative that the
+    retrieved token is included in the `Authorization` header as a `Bearer`
+    token for all subsequent API requests.
+    ## Scopes and Permissions
+    During the token request, developers will need to specify the scopes that
+    correspond to the permissions the application requires. Our API strictly
+    enforces scope validation to ensure clients only perform operations that
+    are within their granted permissions.
+    ## Secure Transmission
+    All communication with the API, including token requests, must occur over
+    HTTPS to ensure data confidentiality and protection.
+    ## Error Handling
+    If a request is made without a token or with an expired or invalid token,
+    the API will return a  `401 Unauthorized` status code. Similarly, if the
+    token does not include the proper scopes for the requested operation,
+    a `403 Forbidden` status code will be returned, indicating
+    insufficient permissions.
+    ---
+  contact:
+    email: sp-edc@lists.camaraproject.org
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+externalDocs:
+  description: Product documentation at CAMARA.
+  url: https://github.com/camaraproject/EdgeCloud
+servers:
+  - url: "{apiRoot}/{basePath}"
+    variables:
+      apiRoot:
+        default: https://localhost:443
+        description: API root.
+      basePath:
+        default: application-endpoint-discovery/vwip
+        description: Base path for the Application Endpoint Discovery.
+tags:
+  - name: Discovery
+    description: |
+      Find the closest App Instance Endpoint to the user device.
+paths:
+  /apps-endpoints:
```

Comment:
`apps-endpoints` or `app-endpoints`? I think it should be app-endpoints, because it is all the endpoints for a single application.. In fact `app-server-endpoints` may be best because we are only returning the server endpoint, not the client endpoint.

Reply:
To me  app-endpoints looks fine as you rightly said it is about the endpoints of the same application. The other one i.e. app-server-endpoints contains "server" keyword which i feel is more concrete and as we are referring mostly the "application" in general so app-endpoints looks more aligned. Any thoughts?

Reply:
Agree